src/pipdeptree/_models/dag.py

```diff
@@ -164,7 +158,9 @@ def filter_nodes(self, include: list[str] | None, exclude: set[str] | None) -> P
                             # a dependency is missing
                             continue
 
-        non_existent_includes = [i for i in include_with_casing_preserved if i.lower() not in matched_includes]
+        non_existent_includes = [
+            i for i in include_with_casing_preserved if pep503_normalize(i) not in matched_includes
+        ]
```

Comment:
Since we changed this to normalize user input now, we should add a test for this to make sure that it does so in the future.

Reply:
Done